06_PDDE_Checker.R

```diff
@@ -437,23 +469,115 @@ overlapping_hmis_participation <- HMISParticipation %>%
   merge_check_info(checkIDs = 131) %>%
   select(all_of(PDDEcols))
 
+
+# Bed Type incompatible with Housing Type -----------------------------------
+#If HousingType is 1 or 2 (site-based), then BedType should be 1 (facility based beds). If HousingType is 3 (tenant-based), then BedType should be 2 (voucher beds).
+
+ES_BedType_HousingType <- Inventory %>%
+  left_join(Project0(), by = "ProjectID") %>%
+  left_join(HousingTypeDF, by = "ProjectID") %>% 
+  filter(ProjectType %in% c(0, 1) &
+           ((HousingType %in% c(1, 2) & ESBedType!=1) | (HousingType==3 & ESBedType!=2)) 
+  ) %>%
+  merge_check_info(checkIDs = 135) %>% 
+  mutate(Detail = 
+           paste0("Bed Type incompatible with Housing Type:  Facility-based beds should align to the Housing Type of site-based and voucher-based beds should align to the Housing Type of tenant-based.")
+  ) %>%
+  select(all_of(PDDEcols))
+
+
+# Project CoC Missing Bed Inventory -----------------------------------
+# Throw a warning if there is no inventory record for a ProjectID and COCCode combo in the ProjectCoC data
+
+ProjectCoC <- ProjectCoC %>% mutate(iy=1)
```

Comment:
Look into the `anti-join` function. I think it's designed for just this purpose, so it'll probably make for cleaner, easier-to-read code, and potentially more efficient processing.